src/model/utils/XrefUtils.tsx

```diff
@@ -183,6 +179,32 @@ export const getUnconditionalImplicitXrefs = () => {
   return foundXrefs;
 };
 
+export const getJoinedXrefs = (xrefs: Xref[]) => {
+  const joinedXrefs = [...xrefs];
+  const scheduledForRemoval: string[] = [];
+  joinedXrefs
+    .filter(xref => xref.properties && xref.properties.Status === 'JOINED')
+    .forEach(xref => {
+      const masterXref = joinedXrefs.find(
```

Comment:
The reference to `masterXref` is getting updated by side effect here. Is it possible to move towards a more functional approach without modifying references?

Reply:
I guess we could make a deep copy of xrefs at the top of the function?

Reply:
It probably doesn't really matter if the xrefs are modified in place (and as an added bonus this will be a quick operation). My only issue really here is really just readability as it's not immediately obvious what happens to `masterXref` until you realise it's being modified in place. I'm happy to leave as is though.